File: backend/src/main/java/de/wps/radvis/backend/wegweisendeBeschilderung/schnittstelle/view/WegweisendeBeschilderungListenView.java
/*
 * Copyright (c) 2023 WPS - Workplace Solutions GmbH
 *
 * Licensed under the EUPL, Version 1.2 or as soon they will be approved by the European Commission - subsequent versions of the EUPL (the "Licence");
 *
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 */

package de.wps.radvis.backend.wegweisendeBeschilderung.schnittstelle.view;

import org.locationtech.jts.geom.Point;

import de.wps.radvis.backend.organisation.schnittstelle.VerwaltungseinheitView;
import de.wps.radvis.backend.wegweisendeBeschilderung.domain.entity.WegweisendeBeschilderung;
import lombok.Getter;

@Getter
public class WegweisendeBeschilderungListenView {
	private final long id;
	private final Point geometrie;
	private final String pfostenNr;
	private final String wegweiserTyp;
	private final String pfostenTyp;
	private final String zustandsbewertung;
	private final String defizit;
	private final String pfostenzustand;
	private final String pfostendefizit;
	private final String gemeinde;
	private final String kreis;
	private final String land;

	private final VerwaltungseinheitView zustaendig;

	public WegweisendeBeschilderungListenView(WegweisendeBeschilderung wegweisendeBeschilderung) {
		this.id = wegweisendeBeschilderung.getId();
		this.zustaendig = new VerwaltungseinheitView(wegweisendeBeschilderung.getZustaendigeVerwaltungseinheit());

		this.geometrie = (Point) wegweisendeBeschilderung.getGeometrie();
		this.pfostenNr = wegweisendeBeschilderung.getPfostenNr().getValue();
		this.wegweiserTyp = wegweisendeBeschilderung.getWegweiserTyp().getValue();
		this.pfostenTyp = wegweisendeBeschilderung.getPfostenTyp().getValue();
		this.zustandsbewertung = wegweisendeBeschilderung.getZustandsbewertung().getValue();
		this.defizit = wegweisendeBeschilderung.getDefizit().getValue();
		this.pfostenzustand = wegweisendeBeschilderung.getPfostenzustand().getValue();
		this.pfostendefizit = wegweisendeBeschilderung.getPfostendefizit().getValue();
		this.gemeinde = wegweisendeBeschilderung.getGemeinde().getValue();
		this.kreis = wegweisendeBeschilderung.getKreis().getValue();
		this.land = wegweisendeBeschilderung.getLand().getValue();
	}
}

File: frontend/radvis/src/app/viewer/wegweisende-beschilderung/models/wegweisende-beschilderung-listen-view.ts
/*
 * Copyright (c) 2023 WPS - Workplace Solutions GmbH
 *
 * Licensed under the EUPL, Version 1.2 or as soon they will be approved by the European Commission - subsequent versions of the EUPL (the "Licence");
 *
 * You may not use this work except in compliance with the Licence.
 * You may obtain a copy of the Licence at:
 *
 * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the Licence for the specific language governing permissions and limitations under the Licence.
 */

import { PointGeojson } from 'src/app/shared/models/geojson-geometrie';
import { Verwaltungseinheit } from 'src/app/shared/models/verwaltungseinheit';
import { NetzDetailFeatureTableLink } from 'src/app/viewer/viewer-shared/models/netzdetail-feature-table-link';

export interface WegweisendeBeschilderungListenView {
  id: number;
  geometrie: PointGeojson;
  pfostenNr: string;
  wegweiserTyp: string;
  pfostenTyp: string;
  zustandsbewertung: string;
  defizit: string;
  pfostenzustand: string;
  pfostendefizit: string;
  gemeinde: string;
  kreis: string;
  land: string;
  zustaendig: Verwaltungseinheit | null;
}

// eslint-disable-next-line @typescript-eslint/no-namespace
export namespace WegweisendeBeschilderungListenView {
  const EMPTY_FIELD_INDICATOR = '';

  export const getDisplayValueForKey = (item: WegweisendeBeschilderungListenView, key: string): string => {
    switch (key) {
      case 'id':
        return `${item.id}` ?? EMPTY_FIELD_INDICATOR;
      case 'pfostenNr':
        return item.pfostenNr ?? EMPTY_FIELD_INDICATOR;
      case 'wegweiserTyp':
        return item.wegweiserTyp ?? EMPTY_FIELD_INDICATOR;
      case 'pfostenTyp':
        return item.pfostenTyp ?? EMPTY_FIELD_INDICATOR;
      case 'zustandsbewertung':
        return item.zustandsbewertung ?? EMPTY_FIELD_INDICATOR;
      case 'defizit':
        return item.defizit ?? EMPTY_FIELD_INDICATOR;
      case 'pfostenzustand':
        return item.pfostenzustand ?? EMPTY_FIELD_INDICATOR;
      case 'pfostendefizit':
        return item.pfostendefizit ?? EMPTY_FIELD_INDICATOR;
      case 'gemeinde':
        return item.gemeinde ?? EMPTY_FIELD_INDICATOR;
      case 'kreis':
        return item.kreis ?? EMPTY_FIELD_INDICATOR;
      case 'land':
        return item.land ?? EMPTY_FIELD_INDICATOR;
      case 'zustaendig':
        return Verwaltungseinheit.getDisplayName(item.zustaendig) ?? EMPTY_FIELD_INDICATOR;
      default:
        throw Error(`Key ${key} nicht gefunden`);
    }
  };

  export const getDetails = (
    item: WegweisendeBeschilderungListenView
  ): { [key: string]: string | NetzDetailFeatureTableLink } => {
    return {
      Pfostennummer: item.pfostenNr,
      Wegweisertyp: item.wegweiserTyp,
      Pfostentyp: item.pfostenTyp,
      Zustandsbewertung: item.zustandsbewertung,
      Defizit: item.defizit,
      Pfostenzustand: item.pfostenzustand,
      Pfostendefizit: item.pfostendefizit,
      Gemeinde: item.gemeinde,
      Kreis: item.kreis,
      Land: item.land,
      'Zuständige Organisation': item.zustaendig?.name ?? '',
      Kataster: {
        text: item.pfostenNr + '.pdf',
        url: '/api/wegweisendebeschilderung/kataster/' + item.pfostenNr + '.pdf',
      },
    };
  };
}
